### DISCUSSION TITLE
Any way to force NPM install packages from npm, not as symlink to local package?

### DISCUSSION THREAD

**Discussion by maksimovsyannikov** [npm/feedback#546] [CLI] [answered]
Hello!
I have a monorepo, using npm monorepo feature. I have a problem like this:

1. There are 2 packages `A` and `B`, `A` depends on `B`
2. I run ci in teamcity. I run `npm i` from root of repository on master branch, if `A` has in dependencies version of `B` which is currently in master branch, npm will install `B` as symlink to local folder. For example `A` depends on `B@1.0.1`, and in master in `B/package.json` I have version `1.0.1`
3. Then I build and publish package `A` from A folder

And here I have problem. I can commit something in master without publishing package `B`. And then in teamcity I will not get the same code of `B@1.0.1` which is published in npm. Because `B` will be installed as symlink

I can't run `npm i` from A folder, because in root package.json I have dev dependencies used for build. I need to run `npm i` from repository root.

I can run `npm i` from root, and then `npm i` from A folder. This way I will get `B` from npm and I will get dev deps. But it looks complicated. Is there an easier way? For example option which will force `npm i` not install local packages as symlinks

**Comment by MylesBorins** [comment, marked as answer]
@maksimovsyannikov what is the use case where you want to install just a single package in the monorepo and need all the dependencies from?

I think you might benefit from putting the expected dev dependencies (and everything else needed for a package to be self contained) into the packages individual package.json rather than having some stuff only at the root.

With all this said, it seems like this is more of a support request than a suggestion for a new feature. This channel is not used for general support, you should open discussion over at https://github.community or stack overflow for support.

If you have a specific feature you'd like to suggest you can open an RFC over at https://github.com/npm/rfcs
